packages/cli/src/menus/shared.ts

```diff
@@ -98,3 +104,98 @@ export async function getAssetAmount(
         { decimals: 0, assetId: assetId },
       );
 }
+
+export async function addressOrHexToHash(
+  address_str: string,
+  expectedType: CredentialType,
+): Promise<string> {
+  if (/[0-9a-fA-F]{56}/.test(address_str)) {
+    return address_str;
+  }
+  if (address_str.startsWith("addr") || address_str.startsWith("stake")) {
+    const address = Address.fromBech32(address_str);
+    if (address_str.startsWith("addr")) {
+      const credType = address.getProps().paymentPart?.type;
```

Comment:
Doesn't this also work for `stake_` addresses? Afaik, it determines these things internally, which is the point of `.getProps()` (so you don't need to know the type).

I could be wrong.

Reply:
I am not sure, this is a function @Quantumplation made for treasury cli (do we need a cli sdk package? :))